packages/dds/tree/src/simple-tree/api/treeNodeApi.ts

```diff
@@ -265,6 +271,11 @@ export const treeNodeApi: TreeNodeApi = {
 				);
 		}
 	},
+	fieldMetadata(node: TreeNode, key: string | number): unknown | undefined {
+		const flexNode = getOrCreateInnerNode(node);
+		const field = flexNode.tryGetField(key as FieldKey);
```

Comment:
This will fail to get metadata for empty fields. Its also an invalid cast when passed in a number.

Also this depends on the flex tree schema layer: please avoid introducing more such dependencies. Use the simple tree schema when writing schema specific logic in the simple tree code.